File: wtplayer/src/WTObject.cpp
#include "basetypes.hpp"
#include "InternalOnLoadCallbackWrapper.hpp"

#include "WTObject.hpp"

WTObject::WTObject(WT* wt_) :
  IUnknown(),
  wt(wt_){
}

WTObject* WTObject::getOwner(){
  APILOG;
  return 0;
}

bool WTObject::getIsValid(){
  APILOG;
  return true;
}

bool WTObject::getIsLoaded(){
  APILOG;
  return true;
}

int WTObject::getErrorNumber(){
  APILOG;
  return -1;
}

void WTObject::setErrorNumber(int Error_Number){
  APILOG;
}

VARIANT WTObject::getUserData(){
  APILOG;
  return this->userdata;
}

void WTObject::setUserData(VARIANT User_Data){
  APILOG;
  //this->userdata = User_Data;
}

bool WTObject::isValid(){
  APILOG;
  return true;
}

bool WTObject::isLoaded(){
  APILOG;
  return true;
}

bool WTObject::isLoadedWithChildren(){
  APILOG;
  return true;
}

void WTObject::setOption(int Option_Number,
                         VARIANT New_Value){
  APILOG;
}

VARIANT WTObject::getOption(int Option_Number){
  APILOG;
  return 0;
}

File: wtplayer/javasrc/wildtangent/webdriver/WTObject.java
package wildtangent.webdriver;

public interface WTObject {
    public WTObject getOwner();

    public boolean isLoadedWithChildren();

    public boolean isValid();

    public void setOnLoad(WTOnLoadEvent var1);

    public boolean getIsValid();

    public int getObjectType();

    public Object getOption(int var1);

    public void setOption(int var1, Object var2);

    public void setOption(int var1, boolean var2);

    public void setOption(int var1, double var2);

    public void setOption(int var1, float var2);

    public void setOption(int var1, int var2);

    //public WTObject duplicate();

    public boolean isLoaded();

    public WT getCreator();

    public void setOnLoadedWithChildren(WTOnLoadEvent var1);

    public int getErrorNumber();

    public void setErrorNumber(int var1);

    public String getName();

    public void setName(String var1);

    public void setUserData(Object var1);

    public Object getUserData();

    public boolean getIsLoaded();
}
